robot/EDA/resources/RelationshipsSettingsPageObject.py

```diff
@@ -21,13 +24,47 @@ def _is_current_page(self):
             error=f"Relationships tab with locator '{locator_tab}' is not available on the page"
         )
 
+    @capture_screenshot_on_error
+    def click_delete_setting_icon(self,**kwargs):
+        """ This method will click the delete icon in edit mode
+            Pass the name of the field to be deleted as keyword arguments from robot file
+        """
+        for field,value in kwargs.items():
+            locator = eda_lex_locators["eda_settings_relationships"]["delete_setting_icon"].format(field,value)
+            self.selenium.wait_until_page_contains_element(locator)
+            self.selenium.wait_until_element_is_visible(locator)
+            self.selenium.double_click_element(locator)
+            time.sleep(1) #This wait is needed for handle alert keyword
+
     def go_to_relationships_sub_tab(self, sub_tab):
         """ Click on the given sub_tab in the Affiliations Settings page """
         locator = eda_lex_locators["relationships_settings_locators"]["sub_tab"].format(sub_tab)
         self.selenium.wait_until_page_contains_element(locator,
                                                        error=f"'{sub_tab}' sub tab is not available on the page")
         self.selenium.click_element(locator)
 
+    def enter_new_reciprocal_setting(self,**kwargs):
+        """ This method will enter the new reciprocal settings
+            Pass the expected value to be entered along with field name as keyword arguments
+        """
+        for field,value in kwargs.items():
+            locator = eda_lex_locators["eda_settings_relationships"]["new_reciprocal_setting"].format(field)
+            self.selenium.wait_until_page_contains_element(locator,
+                                                error=f"'{field}' field is not available ")
+            self.selenium.clear_element_text(locator)
+            self.selenium.get_webelement(locator).send_keys(value)
+
+    def get_total_settings_count(self):
+        """ This method will return the number of rows in the reciprocal settings"""
+        locator = eda_lex_locators["eda_settings_relationships"]["settings_count"]
+        self.selenium.wait_until_page_contains_element(locator, error="Reciprocal settings table is not available")
+        for i in range(3):
+                i += 1
+                actual_count = int(self.selenium.get_element_count(locator))
+                if actual_count > 0 :
+                    break
+        return actual_count
```

Comment:
Why are we looping it ? Does it take a while for the count to update ?

Reply:
Yes.Sometime the DOM loads in the background but the active checkbox takes some time to display it in UI